.gitignore

```diff
@@ -0,0 +1,12 @@
+**/.DS_Store
+**/.idea
+
+**/*.iml
+**/*.log
+**/package-lock.json
+
+**/node_modules
+**/target
+**/dist
```

Comment:
We should also ignore the `coverage/` dir resulting from running `yarn run test -- --coverage`. On this occasion I'd recommend to move (part of) the `.gitignore` file to the `core` package as some of the patterns are only relevant to Node projects, while this repository will contain Maven project(s). Submerging `.gitignore` to the project root will also allow for using absolute paths (e.g. `/target`), which might avoid accidentally ignoring packages inside the project.